redfish-core/lib/pcie_slots.hpp

```diff
@@ -29,6 +32,88 @@
 namespace redfish
 {
 
+inline void doLinkAssociatedDiskBackplaneToChassis(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& chassisId, const std::string& drivePath, size_t index,
+    const std::optional<std::string>& validChassisPath,
+    const std::vector<std::string>& assemblyList)
+{
+    if (!validChassisPath || assemblyList.empty())
+    {
+        BMCWEB_LOG_ERROR("Chassis not found");
+        messages::resourceNotFound(asyncResp->res, "Chassis", chassisId);
+        return;
+    }
+
+    auto it = std::find(assemblyList.begin(), assemblyList.end(), drivePath);
+    if (it == assemblyList.end())
+    {
+        BMCWEB_LOG_ERROR("Drive path {} not found in the assembly list",
+                         drivePath);
+        messages::internalError(asyncResp->res);
```

Comment:
Here we will fall through and set the item using the it (assemblyList.end()) below. Is that correct?

In 1060 we didn't set an item in this case.

Should we be returning here?

Reply:
You're right. It needs to be returned here.